src/climate_data/generate/scenario_daily.py

```diff
@@ -173,113 +177,72 @@ def generate_scenario_daily_main(  # noqa: PLR0912, PLR0915, C901
+    # compute anomaly, resample anomaly and compute scenario data
+    source_key = random.choice(list(source_paths.keys()))
+    variant_paths = random.choice(source_paths[source_key])
+    s_variant = f"{variant_paths[0].stem.split('_')[-1]}"
+    vid = f"{source_key}, Variant : {s_variant}"
+    # load reference (monthly) and target (daily for a given year)
+    try:
+        print(f"{vid}: Loading reference")
+        sref = transform(*[load_variable(vp, "reference") for vp in variant_paths])
+        print(f"{vid}: Loading target")
+        target = transform(*[load_variable(vp, year) for vp in variant_paths])
+
+    except KeyError:
+        print(f"{vid}: Bad formatting, skipping...")
+        return
+
+    print(f"{vid}: computing anomaly")
+    v_anomaly = compute_anomaly(sref, target, anomaly_type)
+
+    print(f"{vid}: resampling anomaly")
+    resampled_anomaly = utils.interpolate_to_target_latlon(v_anomaly, method="linear")
+    print(f"{vid}: computing scenario data")
     if anomaly_type == "additive":
-        scenario_data = historical_reference + ensemble_anomaly.groupby("date.month")
+        scenario_data = historical_reference + resampled_anomaly.groupby("date.month")
     else:
-        scenario_data = historical_reference * ensemble_anomaly.groupby("date.month")
-    scenario_data = scenario_data.drop_vars("month")
-    print("Saving")
+        scenario_data = historical_reference * resampled_anomaly.groupby("date.month")
+
+    print(f"{vid}: Writing draw {draw} from {source_key}-{s_variant}")
     cd_data.save_daily_results(
         scenario_data,
         scenario=cmip6_experiment,
         variable=target_variable,
+        draw=draw,
         year=year,
+        provenance_attribute=f"{source_key}-{s_variant}",
         encoding_kwargs=transform.encoding_kwargs,
     )
 
 
 @click.command()  # type: ignore[arg-type]
 @clio.with_output_directory(DEFAULT_ROOT)
 @clio.with_year(years=clio.VALID_FORECAST_YEARS)
+@clio.with_draw(draws=clio.VALID_DRAWS, allow_all=False)
 @clio.with_target_variable(variable_names=list(TRANSFORM_MAP))
 @clio.with_cmip6_experiment()
 def generate_scenario_daily_task(
-    output_dir: str, year: str, target_variable: str, cmip6_experiment: str
+    output_dir: str, draw: str, year: str, target_variable: str, cmip6_experiment: str
 ) -> None:
-    generate_scenario_daily_main(output_dir, year, target_variable, cmip6_experiment)
+    generate_scenario_daily_main(
+        output_dir, draw, year, target_variable, cmip6_experiment
+    )
 
 
 @click.command()  # type: ignore[arg-type]
 @clio.with_output_directory(DEFAULT_ROOT)
 @clio.with_year(years=clio.VALID_FORECAST_YEARS, allow_all=True)
+@clio.with_draw(draws=clio.VALID_DRAWS, allow_all=True)
 @clio.with_target_variable(variable_names=list(TRANSFORM_MAP), allow_all=True)
 @clio.with_cmip6_experiment(allow_all=True)
 @clio.with_queue()
 @clio.with_overwrite()
 def generate_scenario_daily(
     output_dir: str,
     year: str,
+    draw: str,
```

Comment:
Make sure you're being consistent about argument ordering.  You have both (year, draw, target_variable, cmip6_experiment) and (draw, year, target_variable, cmip6_experiment) in different places.

Reply:
Yeah, sloppy, sorry about that